scripts/mbedtls_framework/code_wrapper/psa_wrapper.py

```diff
@@ -0,0 +1,257 @@
+        'mbedtls_psa_external_get_random', # not a library function
+        'psa_get_key_domain_parameters', # client-side function
+        'psa_get_key_slot_number', # client-side function
+        'psa_key_derivation_verify_bytes', # not implemented yet
+        'psa_key_derivation_verify_key', # not implemented yet
+        'psa_set_key_domain_parameters', # client-side function
+    ]),
+    # PAKE stuff: not implemented yet
+    "not_implemented": frozenset([
+        'psa_crypto_driver_pake_inputs_t *',
+        'psa_pake_cipher_suite_t *',
+    ]),
+    "function_guards": {
+        'mbedtls_psa_register_se_key': 'defined(MBEDTLS_PSA_CRYPTO_SE_C)',
+        'mbedtls_psa_inject_entropy': 'defined(MBEDTLS_PSA_INJECT_ENTROPY)',
+        'mbedtls_psa_external_get_random': 'defined(MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG)',
+        'mbedtls_psa_platform_get_builtin_key': 'defined(MBEDTLS_PSA_CRYPTO_BUILTIN_KEYS)',
+        'psa_crypto_driver_pake_get_cipher_suite' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_password' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_password_len' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_peer' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_peer_len' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_user' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_user_len' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_abort' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_get_implicit_key' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_input' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_output' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_password_key' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_peer' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_role' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_user' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_setup' : 'defined(PSA_WANT_ALG_SOME_PAKE)'
+    }
+}
+
+class PSAWrapper(c_wrapper_generator.Base):
+    """Generate a C source file containing wrapper functions for PSA Crypto API calls."""
+
+    _WRAPPER_NAME_PREFIX = 'mbedtls_test_wrap_'
+    _WRAPPER_NAME_SUFFIX = ''
+
+    _PSA_WRAPPER_INCLUDES = ['<psa/crypto.h>']
+
+    def __init__(self,
+                 output_h_f: str,
+                 output_c_f: str,
+                 in_headers:  Collection[str] = DEFAULTS["input_headers"],
+                 config: Dict[str, Any]= DEFAULTS) -> None:
+
+        super().__init__()
+        self._FUNCTION_GUARDS = super()._FUNCTION_GUARDS.copy()
+        self.in_headers = in_headers
+        self.out_c_f = output_c_f
+        self.out_h_f = output_h_f
+
+        self.mbedtls_root = build_tree.guess_mbedtls_root()
+        self.read_config(config)
+
+        if in_headers:
+            self.read_headers(in_headers)
```

Comment:
An empty `in_headers` isn't a special case here.

```suggestion
        self.read_headers(in_headers)
```

Reply:
I was planning on allowing in_headers being provided with an empty config. The check will be changed to `if in_headers and not config:`